src/coreclr/jit/gentree.cpp

```diff
@@ -16418,11 +16237,6 @@ bool Compiler::gtSplitTree(
                 return false;
             }
 
-            if (node->OperIs(GT_ASG))
-            {
-                return false;
-            }
```

Comment:
Shouldn't this be replaced by a `node->OperIsStore()` check?

Reply:
Ah, I suppose unlike ASG the stores are actually classified properly by `GenTree::IsValue`